CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.11)
+cmake_minimum_required(VERSION 3.9)
```

Comment:
don't need to require newer cmake , with the policy check now in place

Reply:
Ack, my plan was to revert this during the rebase.

I still need to validate the Windows MinGW build. The PR you linked mentions issues with LTO + MinGW.